### PR TITLE
LV-844: Add an after_checkout step for docker/build_image job

### PR DESCRIPTION
**Pull request by aymeric-ledger**
The main purpose is to be able to restore cache between the checkout step and the build docker image step.

### CODE REVIEW

**Comment by rottenbytes** [comment]
Hello, what is the goal for this change ? All changes have to be related to a ticket. I sent you the contributions guidelines via slack.

**Comment by aymeric-ledger** [comment]
The primary goal is to allow to restore cache between the checkout step and docker build image step. I'm doing this while doing [LV-844](https://ledgerhq.atlassian.net/browse/LV-844), altho it's not a ticket specific for cache management.

**Comment by dud225** [comment]
LGTM.
However I'd like to see this ORB be tested before being merged.

**Comment by aymeric-ledger** [comment]
Testing it on https://github.com/LedgerHQ/ledger-vault-hsm-driver/pull/172. Looks like the orb works well, I'll merge once I have finish the CI configuration on this PR.

**Comment by dud225** [comment]
That's not the same orb version
```docker: ledger/docker@dev:add_after_checkout_steps-8d9e52267310493914e9e2c07ddbc6ac66cfc97b```

**Comment by aymeric-ledger** [comment]
@dud225 updated the PR with usage example

File: src/docker/orb.yml [hunk 1]
```diff
@@ -59,6 +59,27 @@ examples:
...
+      A workflow using the `after_checkout` step in `docker/build_image` to restore cache
+      before `docker build` is called.
+    usage:
+      version: 2.1
+
+      orbs:
+        docker: ledger/docker@volatile
+
+      workflows:
+        build_test_and_publish:
```
[the remark this answers is not in the document]
Comment (aymeric-ledger) [reply src/docker/orb.yml]: nice catch 👍

**Pull request merged**